lymph/unilateral.py

```diff
@@ -307,35 +323,38 @@ def growth_probability(self):
         """Return the growth probablities of the lymph
         node levels.
         """
-        return self.growth_edges[0].base_t
+        return self.growth_edges[0].spread_prob
 
     @growth_probability.setter
     def growth_probability(self, new_growth_probability):
         """Set the the growth probablities of the lymph node levels.
         """
         for i, edge in enumerate(self.growth_edges):
-            edge.base_t = new_growth_probability
+            edge.spread_prob = new_growth_probability
 
         if hasattr(self, "_transition_matrix"):
             del self._transition_matrix
 
-
+    # here we could allow for different microscopic parameters for each LNL, but then we would need to
+    # adapt how the microscopic parameter is set and read out. For generality allowing to set a specific
+    # microscopic spread parameter for each LNL would be better. To keep calculation time reasonable,
+    # prohibiting this option would be optimal. What do you think, Roman? (same applies to growth probability)
     @property
     def microscopic_parameter(self) -> float:
         """Return the microscopic spread parameter of the connections between the lymph
-        node levels.
+        node levels which modifies the sprea on edges where the starting node is microscopic
         """
-        return self.lnl_edges[-1].microscopic_parameter
+        return self.lnl_edges[-1].micro_mod
 
     @microscopic_parameter.setter
     def microscopic_parameter(self, microscopic_parameter):
         """Set the microscopic spread probabilities for the connections among the LNLs.
         """
         for i, edge in enumerate(self.lnl_edges):
-            edge.microscopic_parameter = microscopic_parameter
+            edge.micro_mod = microscopic_parameter
         if hasattr(self, "_transition_matrix"):
             del self._transition_matrix
-
+    # the sampling now takes longer because check and assign assigns to spread parameters and microscopic spread. hence we loop through the lnl edges twice. we could optimize that by setting them together! (time increase by 50%!!)

```

Comment:
I think I went over this and did not find a point where the transition matrix is recomputed. 
But I will be putting some more time into the investigation where we gain that extra computation time.

Reply:
No no, leave it for now. Let's make this clean and descriptive first. After that, we'll run a profiler to see where it spends most of the time and then we can optimize. Let's try to avoid premature optimization! 😅